src/tracer.js

```diff
@@ -202,29 +202,40 @@ function getTrimmedMetadata(eventMetadata, isRunner) {
 function getTrimmedTrace(traceSize, jsTrace) {
     let currentTraceSize = traceSize;
     const trimmedTrace = Object.assign({}, jsTrace);
-    trimmedTrace.events = jsTrace.events.sort(event => (['runner', 'trigger'].includes(event.origin) ? -1 : 1));
+    let totalTrimmedExceptions = 0;
+    let totalTrimmedEvents = 0;
+    // Trimming trace exceptions.
+    if (trimmedTrace.exceptions.length > 1) {
+        const firstException = trimmedTrace.exceptions[0];
+        currentTraceSize -=
+          JSON.stringify(trimmedTrace.exceptions).length - JSON.stringify(firstException).length;
+        totalTrimmedExceptions = trimmedTrace.exceptions.length - 1;
+        trimmedTrace.exceptions = [firstException];
+    }
     // Trimming trace events metadata.
-    for (let i = jsTrace.events.length - 1; i >= 0; i -= 1) {
-        const currentEvent = trimmedTrace.events[i];
-        const isRunner = currentEvent.origin === 'runner';
-        let eventMetadata = currentEvent.resource.metadata;
-        if (eventMetadata) {
-            const originalEventMetadataSize = JSON.stringify(eventMetadata).length;
-            const trimmedMetadata = getTrimmedMetadata(eventMetadata, isRunner);
-            if (trimmedMetadata) {
-                eventMetadata = trimmedMetadata;
-                const trimmedSize =
+    if (currentTraceSize >= consts.MAX_TRACE_SIZE_BYTES) {
+        trimmedTrace.events = jsTrace.events.sort(event => (['runner', 'trigger'].includes(event.origin) ? -1 : 1));
+        for (let i = jsTrace.events.length - 1; i >= 0; i -= 1) {
+            const currentEvent = trimmedTrace.events[i];
+            const isRunner = currentEvent.origin === 'runner';
+            let eventMetadata = currentEvent.resource.metadata;
+            if (eventMetadata) {
+                const originalEventMetadataSize = JSON.stringify(eventMetadata).length;
```

Comment:
this is used only in the `if (trimmedMetadata)` block, so move it in.

Reply:
eventMetaData changes inside getTrimmedMetadata (we can fix it by deepcopy of eventMetaData, but for another PR).

Reply:
in that case it's a problematic choice of naming for `getTrimmedMetadata`  (not this PR's fault, but still)